### PR TITLE
Efficient canvas update structure with asynchronous tokio backend

### PR DESCRIPTION
**Pull request by Sytronik**
### Rust backend
- "Tokio runtime" manages image drawing task and image caches asynchronously.
- Whenever receiving image state ( id & ch to draw, sec, width, height, ...) from frontend, it aborts existing drawing task, and starts new drawing task.
- A drawing task does
    1.  cropping existing caches and send them to channel
    2. If caches don't exist, drawing low-resolution version and send them to channel
    3. drawing high-resolution caches, cropping them, and send them to channel
- Whenever frontend requests images, the latest images in channel are returned.

### JS frontend
- It sends image state to backend without receiving images.
- It retrieves images from backend only at every "animation frame" (by using requestAnimationFrame),
so unnecessary canvas update doesn't happen.

### CODE REVIEW

File: src/components/MainViewer/MainViewer.js [hunk 1]
```diff
@@ -47,6 +48,7 @@ function MainViewer({
   const [height, setHeight] = useState(250);
   const drawOptionRef = useRef({px_per_sec: 100});
   const [children, registerChild] = useRefs();
+  const requestRef = React.useRef();
```
Comment (crackyachae) [inline src/components/MainViewer/MainViewer.js]: can take React off
```suggestion
  const requestRef = useRef();
```

File: src/components/MainViewer/MainViewer.js [hunk 2]
```diff
@@ -115,46 +117,43 @@ function MainViewer({
...
+  };
 
-  const throttledDraw = useCallback(throttle(1000 / 60, draw), [draw]);
-  const debouncedDraw = useCallback(debounce(1000 / 30, draw), [draw]);
-  // const throttledDraw = draw;
-  // const debouncedDraw = draw;
+  useEffect(() => {
+    requestRef.current = requestAnimationFrame(draw);
+    return () => cancelAnimationFrame(requestRef.current);
+  }, []);
```
Comment (crackyachae) [inline src/components/MainViewer/MainViewer.js]: Hooks should go to the bottom of the code

**Review by crackyachae — COMMENTED** [review]
minor things to change.

File: src/components/MainViewer/MainViewer.js (hunk 1 above)
Comment (Sytronik) [reply src/components/MainViewer/MainViewer.js]: suggestion accepted.

File: src/components/MainViewer/MainViewer.js (hunk 2 above)
Comment (Sytronik) [reply src/components/MainViewer/MainViewer.js]: suggestion accepted.

**Review by crackyachae — APPROVED** [review]
lgtm

**Pull request merged**